mmv1/third_party/terraform/services/alloydb/resource_alloydb_secondary_cluster_test.go

```diff
@@ -0,0 +1,592 @@
+package alloydb_test
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+// The cluster creation should succeed with minimal number of arguments
+func TestAccAlloydbCluster_secondaryClusterMandatoryFields(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckAlloydbClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAlloydbCluster_secondaryClusterMandatoryFields(context),
+			},
+		},
+	})
+}
+
+func testAccAlloydbCluster_secondaryClusterMandatoryFields(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_alloydb_cluster" "secondary" {
+  cluster_id   = "tf-test-alloydb-secondary-cluster%{random_suffix}"
+  location     = "us-east1"
+  network      = "projects/${data.google_project.project.number}/global/networks/${google_compute_network.default.name}"
+  cluster_type = "SECONDARY"
+
+  continuous_backup_config {
+    enabled = false
+  }
+
+  secondary_config {
+    primary_cluster_name = google_alloydb_cluster.default.name
+  }
+}
+
+resource "google_alloydb_cluster" "default" {
+  cluster_id = "tf-test-alloydb-cluster%{random_suffix}"
+  location   = "us-central1"
+  network    = "projects/${data.google_project.project.number}/global/networks/${google_compute_network.default.name}"
+}
+
+data "google_project" "project" {}
+
+resource "google_compute_network" "default" {
+  name = "tf-test-alloydb-cluster%{random_suffix}"
+}
+
+`, context)
+}
+
+// This test passes if secondary cluster can be created with existing primary cluster and primary instance
+func TestAccAlloydbCluster_secondaryClusterWithPrimaryClusterAndInstance(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckAlloydbClusterDestroyProducer(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAlloydbCluster_secondaryClusterWithPrimaryClusterAndInstanceCreate(context),
+			},
+			{
+				Config: testAccAlloydbCluster_secondaryClusterWithPrimaryClusterAndInstanceRemoveSecondary(context),
+			},
+		},
+	})
+}
+
+func testAccAlloydbCluster_secondaryClusterWithPrimaryClusterAndInstanceCreate(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_alloydb_cluster" "secondary" {
```

Comment:
Can you add tests include more settings on the secondary cluster? We need to ensure that applying any arbitrary configuration will either give the user an error *or* result in a stable configuration with no permadiffs. The current corpus of tests for secondary clusters are not representative of that.